src/CaptureService/include/CaptureService/CaptureServiceBase.h

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef CAPTURE_SERVICE_CAPTURE_SERVICE_BASE_H_
+#define CAPTURE_SERVICE_CAPTURE_SERVICE_BASE_H_
+
+#include <absl/container/flat_hash_set.h>
+#include <absl/synchronization/mutex.h>
+
+#include "CaptureService/ClientCaptureEventCollectorBuilder.h"
+#include "CaptureStartStopListener.h"
+#include "GrpcProtos/capture.pb.h"
+#include "OrbitBase/Result.h"
+#include "ProducerEventProcessor/ClientCaptureEventCollector.h"
+#include "ProducerEventProcessor/ProducerEventProcessor.h"
+
+namespace orbit_capture_service {
+
+// CaptureServiceBase holds common functionality that does not depend on gRPC and can be shared by
+// the platform-specific native orbit capture services and the cloud collector.
+class CaptureServiceBase {
+ public:
+  CaptureServiceBase();
+
+  void AddCaptureStartStopListener(CaptureStartStopListener* listener);
+  void RemoveCaptureStartStopListener(CaptureStartStopListener* listener);
+
+ protected:
+  enum class CaptureInitializationResult { kSuccess, kAlreadyInProgress };
+  [[nodiscard]] CaptureInitializationResult InitializeCapture(
+      ClientCaptureEventCollectorBuilder* client_capture_event_collector_builder);
+  void TerminateCapture();
+
+  void StartEventProcessing(const orbit_grpc_protos::CaptureOptions& capture_options);
+  enum class StopCaptureReason { kClientStop, kMemoryWatchdog };
```

Comment:
Nit: I am not sure if they should be inner classes. But if you go for it (which I would be fine with), I think they should be grouped together and separated by an empty line from the functions.

Reply:
I grouped those enum classes and also added one empty line to separate from the functions.
Thanks for the suggestion!